nomenclature/definition.py

```diff
@@ -112,6 +113,48 @@ def validate(self, df: IamDataFrame, dimensions: list | None = None) -> None:
         ):
             raise ValueError("The validation failed. Please check the log for details.")
 
+    def validate_datetime(self, df: IamDataFrame) -> None:
+        """Validate datetime coordinates against allowed format and/or timezone.
+
+        Parameters
+        ----------
+        df : :class:`pyam.IamDataFrame`
+            Scenario data to be validated against the datetime format.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValueError
+            If `df` fails validation against the required datetime format.
+        """
+        if self.config.time is None:
+            return
+        errors = ErrorCollector()
+        time_column_name = "subannual" if self.config.time.year else "time"

```

Comment:
"subannual" should not be covered in this example but use the standard codelist-validation.